### PR TITLE
[Doc] Scaling limitation 1.13

### PR DESCRIPTION
**Pull request by nadegepepin**

### CODE REVIEW

File: doc/docs/master/reference/scaling_changes.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Pipeline Scaling Limitation in Community Edition
+
+!!! Warning "We are here to help!" 
+    Scaling Pachyderm pipelines is easy to do but hard for users to perfect. 
+
+It can be difficult to balance the breaking down of data science systems into multiple pipelines, horizontally scaling those pipelines to optimize the processing time and effectively understanding Pachyderm's unique data-first approach to development in connecting those pipelines. Lalalal
+
+Starting with this new 1.13.0 release, we introduce a **limit to the number of pipelines and their parallelization** Community Edition users can deploy.
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_changes.md:8]: This phrasing seems clumsy. I wouldn't say it starts with this release, as this is the documentation for the current release and is not included in the others.

File: doc/docs/master/reference/scaling_changes.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+## Scaling Limitation
+
+Our new limitation have been set on:
+
+- The **number of pipelines** deployed: Community Users can deploy **up to 16 pipelines**.
+- The **number of workers** for each pipeline: Community Users can run **up to 8 workers in parallel** on each pipeline.
+
+Our Customer Success Experience has shown us that hitting those limits means that you are likely to be needing our assistance in optimizing your DAGs.
+
+## Lift the limitation
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_changes.md:21]: This is clumsy phrasing.

File: doc/docs/master/reference/scaling_changes.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+We provide an easy way to request a **2 weeks Enterprise Token** that will help you experiment with Pachyderm without limit. //TODO insert a link to the Request token page here
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_changes.md:23]: "2 weeks Enterprise Token" is not proper English.

File: doc/docs/master/reference/scaling_changes.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+!!! note "Please know..." 
+    Pachyderm offers readily available activation keys for proofs-of-concept, startups, academic, nonprofit, or open-source projects. Tell us about your project, get in touch. //TODO Add link to contact sales
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_changes.md:26]: I'm not sure comma is the right separator for two independent sentences.  I might break them into two.

File: doc/docs/master/reference/scaling_changes.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+!!! Note
+    If update pipeline fails for any other reason, it does not log any message related to pipeline limits.
+
+- `pachctl restore` fails when trying to restore with too many pipelines. 
+
+In any case, running any of the following commands below will create a distinct message to stderr as well as to the pachd logs letting you know the number of pipelines limit, the number of pipelines you have, and provide a link to request an Enterprise key.
+
+### Workers limit
+
+When trying to create a parallelism too high:
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_changes.md:50]: This is also clumsy phrasing.

File: doc/docs/master/reference/scaling_changes.md (hunk 3 above)
Comment (pappasilenus) [reply doc/docs/master/reference/scaling_changes.md:23]: I'm also not sure you should mention the precise length of the token.  If we choose to change it, or A/B test different lengths, you'll be changing this document often.

File: doc/docs/master/reference/scaling_changes.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+## What happens when you upgrade to 1.13?
+
+!!! Important
+    All existing pipelines **continue to work** after an upgrade to 1.13. pachd logs will mention the successful startup above the "update pipeline message".
+
+- `pachctl create pipeline` fails as mentionned in [Pipelines limit](#pipelines-limit) above. 
+
+- `pachctl update pipeline` and `pachctl edit pipeline` continue to work **if not creating a new pipeline** and output the message mentioned above.  
+
+- `pachctl extract, list, run, start, stop pipeline` continue to work as before.
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_changes.md:67]: Writing "continue to work as before" assumes the reader knows how they worked before.  I'd rewrite to describe as the software works today, not as we described the changes in design documents.

**Review by pappasilenus — CHANGES_REQUESTED** [review]
I think there are changes to the information architecture--the naming of the documents--as well as to their content as noted. In general, this should describe the state of the system as it works in this release without comparing it to prior releases.  I would simply state what the limits are.  We will provide positioning for the reasons behind the changes in blogposts.  The documentation should simply describe the software.

File: doc/docs/master/reference/scaling_changes.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# Pipeline Scaling Limitation in Community Edition
+
+!!! Warning "We are here to help!" 
+    Scaling Pachyderm pipelines is easy to do but hard for users to perfect. 
+
+It can be difficult to balance the breaking down of data science systems into multiple pipelines, horizontally scaling those pipelines to optimize the processing time and effectively understanding Pachyderm's unique data-first approach to development in connecting those pipelines. Lalalal
```
[the remark this answers is not in the document]
Comment (nadegepepin) [reply doc/docs/master/reference/scaling_changes.md:6]: Renamed scaling_limits.md

File: doc/docs/master/reference/scaling_changes.md (hunk 2 above)
Comment (nadegepepin) [reply doc/docs/master/reference/scaling_changes.md:21]: Removed

File: doc/docs/master/reference/scaling_changes.md (hunk 1 above)
Comment (nadegepepin) [reply doc/docs/master/reference/scaling_changes.md:8]: Copy/Paste from the PRD - Removed -

File: doc/docs/master/reference/scaling_changes.md (hunk 3 above)
Comment (nadegepepin) [reply doc/docs/master/reference/scaling_changes.md:23]: Removing every mention of time for the free trial of the Enterprise Edition.

File: doc/docs/master/reference/scaling_changes.md (hunk 5 above)
Comment (nadegepepin) [reply doc/docs/master/reference/scaling_changes.md:50]: Copy/Paste from PRD.  Rephrasing.

File: doc/docs/master/reference/scaling_limits.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+To experiment with Pachyderm without limit, we provide an easy way to request an [**Enterprise Edition trial token**](https://www.pachyderm.com/trial). Fill in the form (link provided) and receive an enterprise license token.
+
+!!! Info "You might qualify for a free Enterprise license." 
+    Pachyderm offers activation keys for proofs-of-concept, startups, academic, nonprofit, or open-source projects. [Tell us about your project!](https://www.pachyderm.com/trial). 
+
+## What happens when you exceed those limits?
+
+As a general rule, Pachyderm provides a message in the STDERR whenever a limit is encountered.
+
+For example, each `pachctl` command invoking an enterprise feature (such as `pachctl auth,` `pachctl deploy ide`...) logs an alert message in your STDERR with a link to the [Enterprise Edition documentation](https://docs.pachyderm.com/latest/enterprise/) for more information.
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_limits.md:26]: The link is the trial link, not the enterprise link, for all errors.

Pachctl only emits the enterprise link on deploy.

File: doc/docs/master/reference/scaling_limits.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+### Limit on the number of workers per pipeline
+Pachyderm has [two parallelism](https://docs.pachyderm.com/latest/reference/pipeline_spec/#parallelism-spec-optional) strategies: constant and coefficient.
+
+When `constant` parallelism > 8: 
+
+- `pachctl create pipeline` and `pachctl update pipeline` fail. A message to STDERR and pachd logs is generated. You will need to update your pipeline specification file accordingly or activate an Enterprise license.
+
+The previous behavior differs for a `coefficient` parallelism:
+
+- `pachctl create, update, edit pipeline` will succeed. However, Pachyderm will scale the number of workers up to the Community Edition limit, and ignore the remaining workers if the value set on `coefficient` results in more than 8 workers.  A message to STDERR and pachd logs is generated.
```
Comment (pappasilenus) [inline doc/docs/master/reference/scaling_limits.md:52]: It's more accurate to just say a message is logged to the pachd logs.  The stderr logging is for when the user is running the command, where we need to give feedback on why a command failed.

File: doc/docs/master/troubleshooting/pipeline_troubleshooting.md [hunk 10]
```diff
@@ -23,6 +24,38 @@ At the bottom of the document, we'll provide specific troubleshooting steps for
...
+1. Check your number of pipelines and parallelism settings (`"parallelism_spec"` attribute in your pipeline specification files) against our [limits](../../reference/scaling_limits).
+1. Additionally, your stderr and pipeline logs (`pachctl log -p <pipeline name> --master` or `pachctl log -p <pipeline name> --worker`) should contain one or both of those messages:
+    - number of pipelines limit exceeded:
+        ```
+        Pachyderm Community Edition requires an activation key to create more than 16 total pipelines (you have X).  Use the command `pachctl enterprise activate` to enter your key.
+
+        Pachyderm offers readily available activation keys for proofs-of-concept, startups, academic, nonprofit, or open-source projects. Tell us about your project to get one.
+
+        Get a key here:
+        /n <URL> → (Get a Key form)
```
Comment (pappasilenus) [inline doc/docs/master/troubleshooting/pipeline_troubleshooting.md:42]: replace 

```
/n <URL> → (Get a Key form)
```

with the actual url

File: doc/docs/master/troubleshooting/pipeline_troubleshooting.md [hunk 11]
```diff
@@ -23,6 +24,38 @@ At the bottom of the document, we'll provide specific troubleshooting steps for
...
+        ```
+    - max number of workers exceeded:
+        ```
+        This pipeline will only create a total of 8 workers (you specified X). Pachyderm Community Edition requires an activation key to create pipelines with constant parallelism greater than 8. Use the command `pachctl enterprise activate` to enter your key.
+
+        Pachyderm offers readily available activation keys for proofs-of-concept, startups, academic, nonprofit, or open-source projects. Tell us about your project to get one.
+
+        Get a key here:
+        /n <URL  → (Get a Key form)
```
Comment (pappasilenus) [inline doc/docs/master/troubleshooting/pipeline_troubleshooting.md:51]: replace 

```
/n <URL> → (Get a Key form)
```

with the actual url

**Review by pappasilenus — CHANGES_REQUESTED** [review]
Looks good, except for a couple of typos and minor errors.

File: doc/docs/master/enterprise/index.md [hunk 12]
```diff
@@ -1,11 +1,15 @@
...
-purchasing Pachyderm Enterprise, contact our sales team.
+edition of Pachyderm. 
+
+!!! Warning "Get your free-trial token."
+    To try our enterprise features, fill in [this form](https://www.pachyderm.com/trial) or log in to the Pachyderm UI and
+    provide your email address. 
 
+
+If you are interested in
+purchasing Pachyderm Enterprise, contact our sales team.
```
Comment (Nick-Harvey) [inline doc/docs/master/enterprise/index.md:12]: `If you're interested in Pachyderm Enterprise, [schedule some time](https://www.pachyderm.com/request-a-demo/) with one of our experts.

File: doc/docs/master/enterprise/index.md [hunk 13]
```diff
@@ -1,11 +1,15 @@
...
 <div class="btn">
   <div class="center-screen">
     <a href="https://share.hsforms.com/1kC3bWgvMQve902wrlyh8RA2ttwt">Contact Sales</a>
```
Comment (Nick-Harvey) [inline doc/docs/master/enterprise/index.md:15]: `<a href

File: doc/docs/master/reference/scaling_limits.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
+# Pipeline Scaling Limits in Community Edition
+
+Pachyderm Community Edition **limits the number of pipelines** users can deploy and **caps the parallelism** settings of those pipelines.
```
Comment (Nick-Harvey) [inline doc/docs/master/reference/scaling_limits.md:3]: Our completely free Pachyderm Community Edition contains built-in scaling limitations and parallelism thresholds. If you want to scale beyond these limits, request your [FREE 30-Day Pachyderm Enterprise token](https://www.pachyderm.com/trial) today and enjoy unlimited scaling, JupyterHub integration, and more!

You can also enjoy unlimited scaling on [Pachyderm Hub](https://hub.pachyderm.com) if wrangling infrastructure isn't your thing.

File: doc/docs/master/reference/scaling_limits.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+If your DAGs require more than what our Community Edition allows you to run, [ask for a free trial and receive a time-limited enterprise license token](https://www.pachyderm.com/trial). 
+
+## Scaling Limits
+
+|**Number of pipelines** deployed| **Number of workers** for each pipeline|
+|------|------|
+|Community Users can deploy **up to 16 pipelines**.| Community Users can run **up to 8 workers in parallel** on each pipeline.|
+
+Are you hitting those limits? You are likely to be needing our assistance in optimizing your DAGs. We are here to help! See [Lift the limitation](#lift-the-limitation) and request a free trial of our Enterprise Edition.
```
Comment (Nick-Harvey) [inline doc/docs/master/reference/scaling_limits.md:13]: With the addition above we shouldn't need this anymore or the `Lift the Limitations` section, right?

File: doc/docs/master/troubleshooting/pipeline_troubleshooting.md [hunk 16]
```diff
@@ -2,8 +2,9 @@
 
 ## Introduction
 
-Job failures can occur for a variety of reasons, but they generally categorize into 3 failure types: 
+Job failures can occur for a variety of reasons, but they generally categorize into 4 failure types: 
 
+1. You [hit the Community Edition Scaling Limit](#community-edition-scaling-limit).
```
Comment (Nick-Harvey) [inline doc/docs/master/troubleshooting/pipeline_troubleshooting.md:7]: [hit one of the Pachyderm Community Edition Scaling Limits]

**Review by Nick-Harvey — COMMENTED** [review]
Few minor edits

File: doc/docs/master/reference/scaling_limits.md (hunk 14 above)
Comment (nadegepepin) [reply doc/docs/master/reference/scaling_limits.md:3]: Found a common ground with:
Our free Pachyderm Community Edition contains built-in scaling limitations and parallelism thresholds. If you want to scale beyond these limits, request your FREE 30-Day Pachyderm Enterprise token and enjoy unlimited scaling, JupyterHub integration, and more.

**Pull request merged**